### PR TITLE
Form validations

### PR DESCRIPTION
**Pull request by apepper**
This PR adds more validations to ensure, that the form in question is valid. E.g. it ensures, that one field name can only be used once and that a form container is not inside another form container.

Also the final API endpoint is in use now.

Locales are not yet checked by Andreas.

### CODE REVIEW

File: src/Widgets/FormContainerWidget/FormContainerWidgetEditingConfig.js [hunk 1]
```diff
@@ -67,7 +69,15 @@ Scrivito.provideEditingConfig("FormContainerWidget", {
...
+      if (getFormContainer(widget)) {
+        return "Needs to be outside of a form.";
+      }
+
+      if (widgets(widget).every((w) => w.objClass() !== "FormButtonWidget")) {
+        return "A submit button is missing.";
+      }
+
       if (!neoletterInstance()) {
         return "For form containers to work, a Neoletter instance must be specified in the site settings.";
```
Comment (dcsaszar) [inline src/Widgets/FormContainerWidget/FormContainerWidgetEditingConfig.js:82]: * [ ] FYI unrelated idea: While testing I stumbled over the fact that "my first form" is red right from the start. How about moving this check to the submit button? This way, an editor is free to build a mostly working form without having to put dummy data, and an admin can set the API config independently to make the submit buttons work.

**Review by dcsaszar — CHANGES_REQUESTED** [review]
Looks good. Mostly FYI.

File: src/Widgets/FormContainerWidget/validations/typeValidation.js [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
 export const typeValidation = [
   "type",
-  (type) => {
+  (type, { widget }) => {
     if (!type) {
       return "Select the input type.";
     }
+
+    if (!isCustomType(type) && !isFieldNameUnique(widget)) {
+      return "There is at least one other element with the same input type.";
```
[the remark this answers is not in the document]
Comment (apepper) [reply src/Widgets/FormContainerWidget/validations/typeValidation.js:12]: Both use-cases sound like a `custom single-line` but with more browser semantics (e.g. email or phone).

At least for now this is not yet planned.

File: src/Widgets/FormContainerWidget/FormContainerWidgetEditingConfig.js (hunk 1 above)
Comment (apepper) [reply src/Widgets/FormContainerWidget/FormContainerWidgetEditingConfig.js:82]: Interesting idea. I like it.

**Comment by apepper** [comment]
REVIEW (26ab333)

**Review by dcsaszar — APPROVED** [review]

**Pull request merged**